File: QE-selenium/python/Activity11.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.common.action_chains import ActionChains
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

with webdriver.Firefox() as driver:
    driver.get("https://training-support.net/webelements/dynamic-controls")
    print(driver.title)
    checkbox = driver.find_element(By.XPATH,"//input[@id='checkbox']")
    ActionChains(driver).click(driver.find_element(By.XPATH,"//button[contains(text(),'Toggle Checkbox')]")).perform()
    wait = WebDriverWait(driver,10)
    wait.until(EC.invisibility_of_element_located((By.XPATH,"//input[@id='checkbox']")))
    print("Waited Until the checkbox is dissappeared")
    ActionChains(driver).click(driver.find_element(By.XPATH,"//button[contains(text(),'Toggle Checkbox')]")).perform()
    wait.until(EC.visibility_of_element_located((By.XPATH,"//input[@id='checkbox']")))
    print("Waited Until the checkbox is appeared")
    driver.quit()

File: QE-selenium/java/Activity11.java
package demos;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class Activity11 {
    public static void main(String[] args) {

        WebDriver driver = new FirefoxDriver();

        try {
            driver.get("https://training-support.net/webelements/dynamic-controls");
            System.out.println(driver.getTitle());

            Actions actions = new Actions(driver);
            WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));

            
            actions.click(driver.findElement(By.xpath("//button[contains(text(),'Toggle Checkbox')]")))
                   .perform();

            wait.until(ExpectedConditions.invisibilityOfElementLocated(
                    By.xpath("//input[@id='checkbox']"))
            );
            System.out.println("Waited Until the checkbox is dissappeared");

           
            actions.click(driver.findElement(By.xpath("//button[contains(text(),'Toggle Checkbox')]")))
                   .perform();

            wait.until(ExpectedConditions.visibilityOfElementLocated(
                    By.xpath("//input[@id='checkbox']"))
            );
            System.out.println("Waited Until the checkbox is appeared");

        } finally {
            driver.quit();
        }
    }
}